integration-libs/opf/checkout/components/opf-checkout-billing-address-form/opf-checkout-billing-address-form.service.ts

```diff
@@ -0,0 +1,183 @@
+      tap((countries) => {
+        if (Object.keys(countries).length === 0) {
+          this.userPaymentService.loadBillingCountries();
+        }
+      }),
+      // we want to share data with the address form and prevent loading data twice
+      shareReplay(1)
+    );
+  }
+
+  getAddresses(): void {
+    this.isLoadingAddressSub.next(true);
+
+    combineLatest([this.getDeliveryAddress(), this.getPaymentAddress()])
+      .pipe(take(1))
+      .subscribe(
+        ([deliveryAddress, paymentAddress]: [
+          Address | undefined,
+          Address | undefined
+        ]) => {
+          if (!paymentAddress && !!deliveryAddress) {
+            this.setBillingAddress(deliveryAddress);
+            this.billingAddressSub.next(deliveryAddress);
+          }
+
+          if (!!paymentAddress && !!deliveryAddress) {
+            this.billingAddressId = paymentAddress.id;
+            this.billingAddressSub.next(paymentAddress);
+            this.isSameAsDeliverySub.next(false);
+          }
+
+          this.isLoadingAddressSub.next(false);
+        }
+      );
+  }
+
+  putDeliveryAddressAsPaymentAddress(): void {
+    this.getDeliveryAddress()
+      .pipe(
+        switchMap((address: Address | undefined) =>
+          !!address ? this.setBillingAddress(address) : EMPTY
+        ),
+        take(1)
+      )
+      .subscribe({
+        next: () => this.setIsSameAsDeliveryValue(true),
+        complete: () => {},
+        error: () => this.setIsSameAsDeliveryValue(false),
+        // Method is responsible for placing delivery address as a payment address,
+        // so if was not successful, we know for sure that checkbox 'Same as delivery' should be unchecked
+      });
+  }
+
+  setBillingAddress(address: Address): Observable<Address | undefined> {
+    this.isLoadingAddressSub.next(true);
+
+    return this.checkoutBillingAddressFacade
+      .setBillingAddress(this.getAddressWithId(address))
+      .pipe(
+        switchMap(() => {
+          this.activeCartService.reloadActiveCart();
+
+          return this.activeCartService.isStable();
+        }),
+        filter((isStable: boolean) => isStable),
+        switchMap(() => this.getPaymentAddress()),
+
+        tap((billingAddress: Address | undefined) => {
+          if (!!billingAddress && !!billingAddress.id) {
+            this.billingAddressId = billingAddress.id;
+
+            this.billingAddressSub.next(billingAddress);
+            this.opfService.reloadPaymentMode();
+          }
+        }),
+        catchError((error: HttpErrorModel) => {
+          this.globalMessageService.add(
+            { key: 'opf.checkout.errors.updateBillingAddress' },
+            GlobalMessageType.MSG_TYPE_ERROR
+          );
+          return throwError(error);
+        }),
+        finalize(() => {
+          this.isLoadingAddressSub.next(false);
+        }),
+        take(1)
+      );
+  }
+
+  get billingAddressValue(): Address | undefined {
+    return this.billingAddressSub.value;
+  }
```

Comment:
[MEDIUM]

This static getter is not used anywhere. Moreover, we already have an `Observable` interface for this data. Can we remove this static getter?

Note: Analogically, the same applies to static getter `isSameAsDeliveryValue`.

Reply:
Done.